datajunction-clients/python/datajunction/builder.py

```diff
@@ -86,25 +87,67 @@ def restore_namespace(self, namespace: str, cascade: bool = False) -> None:
     #
     # Nodes: all
     #
+    def make_node_of_type(
+        self,
+        type_: models.NodeType,
+        data: Dict,
+    ):
+        """
+        Make a new node of the given type.
+        """
+        # common arguments
+        common_args = [
+            field.name
+            for field in fields(Node)
+            if field.name not in ["dj_client", "type"]
+        ]
+        if type_ == models.NodeType.SOURCE:
```

Comment:
You can probably reduce these to a single call?

Reply:
Ah, you are right. I probably can.